certification/engine/operatorsdk.go

```diff
@@ -0,0 +1,182 @@
+		return nil, fmt.Errorf("%w: %s", errors.ErrOperatorSdkScorecardFailed, err)
+	}
+	scorecardData.Stdout = stdout.String()
+	scorecardData.Stderr = stderr.String()
+	return &scorecardData, nil
+}
+
+func (o operatorSdkEngine) BundleValidate(image string, opts cli.OperatorSdkBundleValidateOptions) (*cli.OperatorSdkBundleValidateReport, error) {
+	cmdArgs := []string{"bundle", "validate"}
+	if opts.ContainerEngine == "" {
+		opts.ContainerEngine = "podman"
+	}
+	cmdArgs = append(cmdArgs, "-b", opts.ContainerEngine)
+	if opts.OutputFormat == "" {
+		opts.OutputFormat = "json-alpha1"
+	}
+	cmdArgs = append(cmdArgs, "--output", opts.OutputFormat)
+	if opts.Selector != nil {
+		for _, selector := range opts.Selector {
+			cmdArgs = append(cmdArgs, "--select-optional", fmt.Sprintf("name=%s", selector))
+		}
+	}
+	if opts.Verbose {
+		cmdArgs = append(cmdArgs, "--verbose")
+	}
+	cmdArgs = append(cmdArgs, image)
+
+	cmd := exec.Command("operator-sdk", cmdArgs...)
+	log.Debugf("Command being run: %s", cmd.Args)
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+	if err != nil {
+		log.Debugf("Stderr: %s", stderr.String())
+		return &cli.OperatorSdkBundleValidateReport{Stderr: stderr.String()}, fmt.Errorf("%w: %s", errors.ErrOperatorSdkBundleValidateFailed, err)
+	}
+
+	var bundleValidateData cli.OperatorSdkBundleValidateReport
+	if strings.Contains(opts.OutputFormat, "json") {
+		err = json.Unmarshal(stdout.Bytes(), &bundleValidateData)
+		if err != nil {
+			return nil, fmt.Errorf("%w: %s", errors.ErrOperatorSdkBundleValidateFailed, err)
+		}
+	} else {
+		if strings.Contains(stdout.String(), "ERRO") || strings.Contains(stdout.String(), "FATA") {
+			bundleValidateData.Passed = false
+		}
+	}
+	bundleValidateData.Stdout = stdout.String()
+	bundleValidateData.Stderr = stderr.String()
+
+	return &bundleValidateData, nil
+}
+
+func (o operatorSdkEngine) writeScorecardFile(resultFile, stdout string) error {
+	scorecardFile := fileutils.ArtifactPath(resultFile)
+
+	err := ioutil.WriteFile(scorecardFile, []byte(stdout), 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func createScorecardConfigFile() (string, error) {
+	configTemplate := `kind: Configuration
+	apiversion: scorecard.operatorframework.io/v1alpha3
+	metadata:
+	  name: config
+	stages:
+	  - parallel: true
+		tests:
+		  - image: quay.io/operator-framework/scorecard-test:v1.9.0
+			entrypoint:
+			  - scorecard-test
+			  - basic-check-spec
+			labels:
+			  suite: basic
+			  test: basic-check-spec-test
+		  - image: quay.io/operator-framework/scorecard-test:v1.9.0
+			entrypoint:
+			  - scorecard-test
+			  - olm-bundle-validation
+			labels:
+			  suite: olm
+			  test: olm-bundle-validation-test
+`
+
+	tempConfigFile, err := os.CreateTemp("", "scorecard-test-config-*.yaml")
+	if err != nil {
+		log.Error("could not create temp config file", err)
+		return "", err
+	}
+	_, err = tempConfigFile.WriteString(configTemplate)
```

Comment:
I am wondering If the tempConfigFile has been created successfully, but the function fails to run `WriteString`, do we want to return tempConfigFile as non-nil value along with a non-nil error?

Reply:
nvm; Found out how the caller is making use of both return values.